packages/@expo/cli/src/prebuild/prebuildAsync.ts

```diff
@@ -65,6 +65,19 @@ export async function prebuildAsync(
   setNodeEnv('development');
   require('@expo/env').load(projectRoot);
 
+  const { platforms } = getConfig(projectRoot).exp;
+  if (platforms?.length) {
+    // Filter out platforms that aren't in the app.json.
+    const finalPlatforms = options.platforms.filter((platform) => platforms.includes(platform));
+    if (finalPlatforms.length > 0) {
+      options.platforms = finalPlatforms;
+    } else {
+      Log.warn(
+        chalk`⚠️ No platforms to prebuild, skipping. Provided platforms not present in app.json.`
```

Comment:
This is an unlikely error, but let's make it better: please include the "provided platforms" and those present in app.json into the error message so that user understands it better.

Also, `app.json` isn't the only file where configuration happens - people might be using `app.config.js|ts` - let's call it "app config" as it's called in other places.

Reply:
I've made requested changes.